Plugins/NIOSoundness/Soundness.swift

```diff
@@ -0,0 +1,251 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+#if swift(>=5.7)
+import PackagePlugin
+import Foundation
+
+@main
+@available(macOS 13.0, *)
+struct Soundness: CommandPlugin {
+    func performCommand(context: PackagePlugin.PluginContext, arguments: [String]) async throws {
+        // Check for help first.
+        if arguments.contains("--help") {
```

Comment:
SwiftPM plugin APIs include a simple argument parser that can help make these tasks easier. cc @abertelrud